### PR TITLE
AMP code cleaning: ArticleController and LiveBlogController updates

### PR DESCRIPTION
**Pull request by shtukas**
## What does this change?

1. Update the `RenderType` type wto discriminate between `LocalRenderArticle` and `LocalRenderAmp`, this to help reasoning about the AMP rendering. Then update the two controllers `ArticleController` and `LiveBlogController`

2. Get rid of switch `DotcomRenderingAMP`.

3. Get rid of attribute `guuiOptOut`

@MatthewJWalls :)

### CODE REVIEW

File: article/app/services/dotcomponents/ArticlePicker.scala [hunk 1]
```diff
@@ -134,7 +134,7 @@ object ArticlePicker {
     val isWhitelisted = whitelist(page.metadata.id)
     val isEnabled = conf.switches.Switches.DotcomRendering.isSwitchedOn
 
-    val tier = if ((isSupported && isWhitelisted && isEnabled && !request.guuiOptOut) || request.isGuui) RemoteRender else LocalRender
+    val tier = if ((isSupported && isWhitelisted && isEnabled) || request.isGuui) RemoteRender else LocalRenderArticle
```
Comment (nicl) [inline article/app/services/dotcomponents/ArticlePicker.scala]: I think the confusing thing is that, at this point, we're not really judging whether to remote render AMP, but whether to render AMP at all for this particular content/request.

Comment (shtukas) [reply article/app/services/dotcomponents/ArticlePicker.scala]: Thanks! Addressing now

File: article/app/controllers/ArticleController.scala [hunk 2]
```diff
@@ -83,14 +79,16 @@ class ArticleController(contentApiClient: ContentApiClient, val controllerCompon
...
+    val tier = ArticlePicker.getTier(article)
+    val isAmpSupported = article.article.content.shouldAmplify
+    request.getRequestFormat match {
+      case JsonFormat if request.isGuui => Future.successful(common.renderJson(getGuuiJson(article, blocks), article).as("application/json"))
+      case JsonFormat => Future.successful(common.renderJson(getJson(article), article))
+      case EmailFormat => Future.successful(common.renderEmail(ArticleEmailHtmlPage.html(article), article))
+      case HtmlFormat if tier == RemoteRender => remoteRenderer.getArticle(ws, path, article, blocks)
+      case HtmlFormat => Future.successful(common.renderHtml(ArticleHtmlPage.html(article), article))
+      case AmpFormat if isAmpSupported => remoteRenderer.getAMPArticle(ws, path, article, blocks)
+      case AmpFormat => Future.successful(common.renderHtml(ArticleHtmlPage.html(article), article))
```
Comment (nicl) [inline article/app/controllers/ArticleController.scala]: Perhaps we should just 404 here?

**Review by nicl — APPROVED** [review]

File: article/app/controllers/ArticleController.scala (hunk 2 above)
Comment (shtukas) [reply article/app/controllers/ArticleController.scala]: https://trello.com/c/RcycqQxB/640-should-we-404-non-ampable-articles-or-serve-them-as-html

**Review by MatthewJWalls — APPROVED** [review]

**Pull request merged**